features/features.go

```diff
@@ -0,0 +1,75 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package features
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// SnapdFeature is a named feature that may be on or off.
+type SnapdFeature int
+
+const (
+	// Layouts controls availability of snap layouts.
+	Layouts SnapdFeature = iota
+	// ParallelInstances controls availability installing a snap multiple times.
+	ParallelInstances
+	// Hotplug controls availability of dynamically creating slots based on system hardware.
+	Hotplug
+	// SnapdSnap controls possibility of installing the snapd snap.
+	SnapdSnap
+	// PerUserMountNamespace controls the persistence of per-user mount namespaces.
+	PerUserMountNamespace
+	// PerformanceMeasurements controls the collection of performance data.
```

Comment:
We don't have this yet, I think we should not add it yet.

Reply:
Sure